src/renderer/kube-object.store.ts

```diff
@@ -195,10 +195,9 @@ export abstract class KubeObjectStore<T extends KubeObject = any> extends ItemSt
     const items = this.items.toJS();
 
     for (const {type, object} of this.eventsBuffer.clear()) {
-      const { uid, selfLink } = object.metadata;
-      const index = items.findIndex(item => item.getId() === uid);
+      const index = items.findIndex(item => item.getId() === object.metadata?.uid);
       const item = items[index];
-      const api = apiManager.getApi(selfLink);
+      const api = apiManager.getApiByKind(object.kind, object.apiVersion);
```

Comment:
Probably this required for new kube-objects coming from watch-apis? (since `watch-api` based on different transport the receiving kube objects coming raw/not yet fixed?)

```ts
const newItem = new api.objectConstructor(object);
api.ensureObjectSelfLink(newItem); // apply fix?
```

Reply:
I think you are right, integration tests fail when an object comes from watch-api.

Reply:
Fixed.